lib/nitra/workers/cucumber.rb

```diff
@@ -39,7 +48,26 @@ def run_file(filename, preloading = false)
           "parts_to_run"  => runnable_parts,
         }
       else
-        run_with_arguments("--no-color", "--require", "features", filename)
+        if configuration.cucumber_format && !preloading
+          args << '--format'
+          args << 'pretty'
+          args << '--format'
+          args << configuration.cucumber_format
+        end
+
+        if configuration.cucumber_out && !preloading
+          args << '--out'
+
+          if filename.include?(':')
+            args << File.join(configuration.cucumber_out, Pathname(filename).sub_ext('').to_s + '_' + filename.partition(':').last)
```

Comment:
Same potential for refactoring into a method with the intent more clearly expressed.